src/transport_ofi.c

```diff
@@ -86,6 +87,16 @@ static char                     myephostname[EPHOSTNAMELEN];
 #endif
 #ifdef ENABLE_THREADS
 shmem_internal_mutex_t          shmem_transport_ofi_lock;
+/* Need a syscall to gettid() because glibc doesn't provide a wrapper
+ * (see gettid manpage in the NOTES section): */
+    #ifndef __APPLE__
+        #include <sys/syscall.h>
+        #ifdef SYS_gettid
+        #define gettid() syscall(SYS_gettid)
+        #else
+        #warning "SYS_gettid not available"
```

Comment:
This should probably be ```#error``` since the build will fail anyway when ```gettid()``` is undefined.  Also, ```#error``` is standard C, but ```#warning``` is an extension.